package_test.go

```diff
@@ -1165,40 +1165,6 @@ func (s *PackageSuite) TestPreparedStmtCaching(c *C) {
 	checkCacheEmpty()
 }
 
-func (s *PackageSuite) TestTransactionWithOneConn(c *C) {
```

Comment:
I think we should have some protection against this still. I have a few ideas but I'm not sure about any of them.
1. In `NewDB` we return an  error if the max connections is 1. The problem with this is that a deadlock can still if the number of in progress transactions matches the number connections in the pool. Also the user can still increase the max connections.
2. We could track the number of connections ourselves and have a sqlair version of `SetMaxOpenConns`. Then we could not prepare on the DB if we know it would block. Again, the problem with this is that the user can still use the "plain" DB and we would not control it. A solution for this would be to also override the Open method and remove `PlainDB` however this would be a breaking change.

Reply:
Regarding 2. I don't think that is the responsibility of SQLair, it should be done at the sql std library instead. Essentially we would be having a parallel "pool" implementation/accounting.